wgpu/src/api/device.rs

```diff
@@ -80,28 +80,30 @@ impl Device {
     pub fn create_shader_module(&self, desc: ShaderModuleDescriptor<'_>) -> ShaderModule {
         let module = self
             .inner
-            .create_shader_module(desc, wgt::ShaderBoundChecks::new());
+            .create_shader_module(desc, wgt::ShaderRuntimeChecks::checked());
         ShaderModule { inner: module }
     }
 
-    /// Creates a shader module from either SPIR-V or WGSL source code without runtime checks.
+    /// Creates a shader module.
     ///
     /// # Safety
-    /// In contrast with [`create_shader_module`](Self::create_shader_module) this function
-    /// creates a shader module without runtime checks which allows shaders to perform
-    /// operations which can lead to undefined behavior like indexing out of bounds, thus it's
-    /// the caller responsibility to pass a shader which doesn't perform any of this
+    ///
+    /// In contrast with [`create_shader_module`][csm] this function
+    /// creates a shader module with user-customizable runtime checks which allows shaders to
+    /// perform operations which can lead to undefined behavior like indexing out of bounds,
+    /// thus it's the caller responsibility to pass a shader which doesn't perform any of this
     /// operations.
     ///
-    /// This has no effect on web.
+    /// When running on WebGPU, this will merely call [`create_shader_module`][csm].
+    ///
+    /// [csm]: Self::create_shader_module
     #[must_use]
     pub unsafe fn create_shader_module_unchecked(
```

Comment:
It's not all that clear to me why this is still unsafe.

Reply:
Why so? The idea gets is that while the ShaderRuntimeChecks constructors are unsafe, that's some serious spooky action at a distance, especially if the ShaderRuntimeChecks type comes from elsewhere in user code. Maybe then that's an argument for just making create_shader_module_unchecked unsafe, and ShaderRuntimeChecks creation safe.

Reply:
I think that way around makes the most sense, of this function being unsafe but the settings object being safe to create.

Reply:
Changed as described